test/dynamo/test_aot_autograd_cache.py

```diff
@@ -87,6 +87,43 @@ def fn(x, y):
         self.assertEqual(counters["aot_autograd"]["autograd_cache_hit"], 1)
         self.assertEqual(counters["aot_autograd"]["autograd_cache_saved"], 1)
 
+    @inductor_config.patch("fx_graph_remote_cache", False)
+    @inductor_config.patch("fx_graph_cache", True)
+    @functorch_config.patch({"enable_autograd_cache": True})
+    def test_activation_checkpointing(self):
+        """
+        Verify that AC work
+        """
+
+        def gn(x, y):
+            return torch.sigmoid(torch.matmul(x, y))
+
+        def fn(x, y):
+            return torch.utils.checkpoint.checkpoint(
+                gn, torch.sin(x), y, use_reentrant=True
+            )
```

Comment:
Right so the case I'm not sure about is: if you pass it two different context_fn, then it should cache miss. Do we implement this correctly?